app/helpers/settings_helper.rb

```diff
@@ -3,8 +3,10 @@ def value(setting)
     return readonly_field(setting, :value,
       {:title => _("This setting is defined in the configuration file 'settings.yaml' and is read-only."), :helper => :show_value}) if setting.readonly?
 
-    return edit_select(setting, :value,
-      {:select_values => self.send("#{setting.name}_collection").to_json }) if self.respond_to? "#{setting.name}_collection"
+    if self.respond_to? "#{setting.name}_collection"
+      return edit_select(setting, :value,
```

Comment:
The select element in setting (Jeditable jquery plugin) lacking in optgroup feature, which can suit here.